pkg/prowgen/prowgen.go

```diff
@@ -399,6 +399,39 @@ func InitializeOpenShiftReleaseRepository(ctx context.Context, openShiftRelease
 	if err := GitCheckout(ctx, openShiftRelease, "master"); err != nil {
 		return err
 	}
+
+	// Remove yaml files for old branches by generating old branch candidates that have a smaler version compared to the minimum
+	// version in the repo branch config. The assumption is that branches are of the form release-vX.Y.
+	for _, branchPrefix := range getOldBranchCandidatePrefixes(inConfig.Config.Branches, "release-v") {
```

Comment:
S-O not covered yet but we could as soon as we decide what branches we want to keep. For now leaving it untouched although if we remove the related midstream branches the corresponding S-O ones should follow.